internal/proto/encoding.go

```diff
@@ -253,3 +333,47 @@ func MarshalToRepeatedBytes[T Bytes](b []byte, num protowire.Number, v []T) int
 	}
 	return off
 }
+
+// SizeRepeatedMessages returns the encoded size of 'repeated M' protobuf field
+// with given number and values.
+func SizeRepeatedMessages[T any, M interface {
+	*T
+	Message
+}](num protowire.Number, v []M) int {
+	var sz int
+	var zero T
+	for i := range v {
+		if !isMessageNil(v[i]) {
+			sz += SizeEmbedded(num, v[i])
+		} else {
+			sz += SizeEmbedded(num, M(&zero))
```

Comment:
i do not understand, it is a really complicated generic func that just tries to handle zero value in a generic fashion? still do not understand why it is needed since https://github.com/nspcc-dev/neofs-sdk-go/pull/667#discussion_r1898363543 thread

Reply:
for the sake of callers - they directly pass codegened fields for encoding. This allowed to drop `for` loop for all `repeated` fields. Btw, tbh, it this rly so complex func?

Reply:
> Btw, tbh, it this rly so complex func?

1 because of its definition
```
func SizeRepeatedMessages[T any, M interface {
	*T
	Message
}](num protowire.Number, v []M) int {
```
2 because of magic `T any` generic type. it does not do what is usually done about generics like "i want certain types here", it does not take part in args

Reply:
i tried my best to generalize `[]*T` args and encode `repeated` fields in one line of code. And i agree that definition is not very convenient to read. May u pls suggest more readable definition?